rust/core-lib/src/editor.rs

```diff
@@ -256,12 +256,15 @@ impl<W: Write + Send + 'static> Editor<W> {
         let text = Rope::from(&edit.text);
         let rev_len = self.engine.get_rev(edit.rev as usize).unwrap().len();
         let delta = Delta::simple_edit(interval, text, rev_len);
+        let prev_head_rev_id = self.engine.get_head_rev_id();
         //self.engine.edit_rev(0x100000, undo_group, edit.rev as usize, delta);
         self.engine.edit_rev(edit.priority as usize, undo_group, edit.rev as usize, delta);
         self.text = self.engine.get_head();
-        if edit.after_cursor {
-            //FIXME: this needs to take account of potential changes between this rev and head.
-             self.new_cursor = Some((edit.start as usize, edit.end as usize));
+
+        // adjust cursor position so that the cursor is not moved by the plugin edit
+        let (changed_interval, _) = self.engine.delta_rev_head(prev_head_rev_id).summary();
```

Comment:
At this line I'd like some feedback, because there is maybe a better way of retrieving the data I am looking for. I'll try to explain what I want.

Lets consider the following scenario (cursor is represented by | ) :

1. `(|` add opening parenthesis which will trigger the plugin
2. `(asdf|` continue writing
3. `(asdf|)` closing parenthesis arrives from the plugin

In this case `edit.start` will be `1`, but I am already at column `5`. `self.engine.edit_rev` properly integrates the changes, taking my intermediate edits into account. However, for the if statement below, I need the actual edit interval of the plugin edit for the current head revision. This is why I retrieve the delta between the previous and newly created revision head here.
In other words: The edit is at interval `(1, 1)`, but it is in the end applied at `(5, 5)`, and `(5, 5)` is what I need to know.

This is probably hard to follow, so if it is impossible to understand please tell me and I'll try to rephrase it.